tests/settings/merged/__init__.py

```diff
@@ -16,6 +16,9 @@
     # Missing file:
     optional('components/missing_file.py'),
 
+    # Missing compiled file
+    optional(compiled("components/missing_file.pyc")),
```

Comment:
I think that we need to test this more. Can you please create a new test with existing `optional(compiled('file'))` and `compiled(optional('file'))`?

What do you think: should we allow arbitary nesting? Or should we only use one way? Or maybe `optional_compiled` function?

Reply:
I think `optional(compiled())` makes sense because it indicates that a compiled-file may optionally be present.

The opposite nesting `compiled(optional())` seems to imply that a file may or may not be present but has to be compiled? That's not making sense to me.

Also `compiled(optional(())` doesn't work because it loses the wrapping of `_Optional` once it is wrapped by `_Compiled`.

Reply:
Please, don't forget to document that.